client/web/src/enterprise/batches/list/BatchChangeListFilters.tsx

```diff
@@ -1,37 +1,85 @@
+    return str.charAt(0).toUpperCase() + str.slice(1).toLowerCase()
+}
 
-interface BatchChangeListFiltersProps
-    extends Required<Pick<MultiSelectProps<MultiSelectOption<BatchChangeState>>, 'onChange' | 'value'>> {
+interface BatchChangeListFiltersProps {
+    filters: BatchChangeState[]
+    selectedFilters: BatchChangeState[]
+    onFiltersChange: (filters: BatchChangeState[]) => void
     className?: string
-    isExecutionEnabled: boolean
 }
 
-export const BatchChangeListFilters: React.FunctionComponent<React.PropsWithChildren<BatchChangeListFiltersProps>> = ({
-    isExecutionEnabled,
-    ...props
-}) => (
-    <>
-        {/* TODO: This should be a proper label. MultiSelect currently doesn't support that being inline though, so this is for later. */}
-        <H4 as={H3} className="mb-0 mr-2">
-            Status
-        </H4>
-        <MultiSelect
-            {...props}
-            options={isExecutionEnabled ? STATUS_OPTIONS : STATUS_OPTIONS_NO_DRAFTS}
-            aria-label="Select batch change status to filter."
-        />
-    </>
-)
+export const BatchChangeListFilters: FC<BatchChangeListFiltersProps> = props => {
+    const { filters, selectedFilters, onFiltersChange, className } = props
+
+    const id = useId()
+    const [searchTerm, setSearchTerm] = useState('')
+
+    // Render only non-selected filters and filters that match with search term value
+    const suggestions = filters.filter(filter => !selectedFilters.includes(filter) && filter.includes(searchTerm))
+
+    return (
+        <Label htmlFor={id} className={classNames(className, styles.root)}>
+            <H4 as={H3} className="mb-0 mr-2">
+                Status
+            </H4>
+
+            <MultiCombobox
+                selectedItems={selectedFilters}
+                getItemName={format}
+                getItemKey={format}
+                onSelectedItemsChange={onFiltersChange}
+                aria-label="Select batch change status to filter."
+            >
+                <MultiComboboxInput
+                    id={id}
+                    value={searchTerm}
+                    placeholder="Select filter..."
+                    onChange={event => setSearchTerm(event.target.value)}
+                />
+
+                <MultiComboboxPopover>
+                    <MultiComboboxList items={suggestions}>
+                        {filters =>
+                            filters.map((filter, index) => (
+                                <MultiComboboxOption key={filter.toString()} value={format(filter)} index={index} />
+                            ))
+                        }
+                    </MultiComboboxList>
+
+                    {suggestions.length === 0 && (
+                        <span className={styles.noFilters}>
+                            {!searchTerm ? (
+                                <>All filters are selected, there are no any other filters</>
```

Comment:
Maybe just this is sufficient

```suggestion
                                <>All filters are selected</>
```